### ISSUE TITLE
TAStudio: Desyncs can still occur when painting upwards

### ISSUE THREAD

**Issue by RetroEdit** [TASVideos/BizHawk#2351]
I've seen this three times: the topmost frames in the painted section are not properly greenzoned (they appear as white instead of being marked by their lag color). I wish I had more than that, but I haven't tried to debug this yet, and it's somewhat inconsistent in reproducing.

This is subtle enough that it may not be fixable in time for 2.5. I'll look into it more later if I get the chance.

Unfortunately, this means that #2186 is not completely fixed (or maybe the root cause wasn't fixed, and it was just obscured).

**Comment by vadosnaprimer** [comment]
Autorestore on or off? Turbo? Follow cursor? Does state gap matter?

**Comment by RetroEdit** [comment]
Autorestore off, follow cursor on, there was no state gap in the location I was painting.

But these just happen to be the conditions the latest time it occurred, they aren't necessarily ways of narrowing down when the problem occurred. For instance, in other similar situations, I've probably had follow cursor off. Sure, it *could* help, but my hope is to be able to provide a concrete example to demonstrate how it works so it can actually be reproduced.

**Comment by RetroEdit** [comment]
My working theory is that one of the preconditions for seeking to the topmost (earliest drawn) frame is not being met after MouseUp for the left mouse button is received.

This is based on some debugging I did last week for a related issue. The problem seems to be related to what I saw this code doing.

https://github.com/TASVideos/BizHawk/blob/master/src/BizHawk.Client.EmuHawk/tools/TAStudio/TAStudio.ListView.cs#L833-L856

If my theory is true, either `CurrentTasMovie.LastEditedFrame` is not properly being set to the topmost (i.e. earliest) drawn frame, or `_startBoolDrawColumn` is somehow wrong? I think it's unlikely `AxisEditingMode` is erroneously set, since I'm not doing anything with axis controls at all. `CurrentTasMovie.LastEditedFrame` seems the most likely culprit.

MouseUp is almost certainly being received, because I can hold the mouse button down a while and playback resumes without doing the proper seeking back.
